woocommerce/woocommerce-ios
# Edit Product: update price settings remotely

Closes #1423 

This is the latest PR for closing #1423. 🎉 

## Description
This PR enabled updating the Product price settings remotely.
Also, this fixes a problem on format value which can compromise the pricing showed in some screens when the website uses a decimal separator which is different from the locale decimal separator.

## Changes
- When encoding a `Product` in the Networking layer, included the 6 pricing properties.
- Configured a date encoding strategy when we try to serialize a model to a dictionary.
- Updated the method `formatAmount()` in `CurrencyFormatter` which initially tried to convert a string to decimal without considering the locale decimal separator. Added also two tests for the format amount method.
- `DefaultProductFormTableViewModel` now use the regular price if available. 
- Passed a completion block to `ProductPriceSettingsViewController` when the user taps "Done" on the Price Settings screen
- Added `priceSettingsUpdated` to ProductUpdater protocol, and implemented it in the Product extension
- Updated Product's price settings locally in `ProductFormViewController` when the user completes updating the price settings
- Updated unit tests

## Testing 
1) Launch the app
2) Go to Products tab
3) Tap on a simple Product
4) Tap "Edit" in the navigation bar
5) Tap the Price settings row
6) Play with all the price settings
7) Tap "Done" --> the fields in the Price settings row should be updated (the sale period will be shown when this PR https://github.com/woocommerce/woocommerce-ios/pull/1694 is merged)
8) Tap "Done" on the Edit Product main screen --> the Product price settings should be updated on wp-admin
If possible, repeat the above steps with different prices and values and confirm the properties are updated on wp-admin.

 
Update release notes:
